test/hotspot/jtreg/serviceability/jvmti/events/FramePop/framepop02/libframepop02.cpp

```diff
@@ -213,23 +215,25 @@ void JNICALL FramePop(jvmtiEnv *jvmti, JNIEnv *jni,
                       jthread thr, jmethodID method, jboolean wasPopedByException) {
   jint frameCount;
 
-  RawMonitorLocker rml(jvmti, jni, agent_lock);
+  if (!isTestThread(jni, jvmti, thr)) {
+    return; // not a tested thread
```

Comment:
If I read the test correctly, NotifyFramePop will only be called with the test thread so therefore the FramePop callback should only be called by the test thread, is that correct? I'm asking because I'm wondering why FramePop also checks the thread is the test thread.

Reply:
As I understand, it is kind of a double-check in this test.
I agree, we can simplify this code little bit.